### PR TITLE
feat(content-switcher): adds fade up animation

### LINKED ISSUE #10117: Content Switcher: Adds micro-interaction
### Summary

Our current Content switcher does not have motion. We will like to add the "Fade up" motion into the content switcher. 

🔗[Link to prototype](https://codesandbox.io/s/content-switcher-motion-carbon-vanilla-on16c?file=/src/index.js)🔗

https://user-images.githubusercontent.com/12755042/141836505-6705fc12-69db-4e45-9f1c-916a81ab06f2.mov

Motion designer contacts on Slack:
- Andrew Smith @ampersandrew

### Justification

As we all know, motion in our product gives the user a sense of control, engages the user, it allows for fluid flows which result in a better user experience. It is also a strong indicator that an action is in progress. 

### Desired UX and success metrics

_No response_

### Required functionality

_No response_

### Specific timeline issues / requests

_No response_

### Available extra resources

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by AlexanderMelox**
Closes #10117 

https://user-images.githubusercontent.com/12755042/142495678-8444683b-f1ff-40f7-aaaf-cc46f3b8e177.mov

This PR adds a fade-up micro-interaction to the Content Switcher component. This change only changes how the background appears behind the content switcher. No additional functionality was changed in the component. 

#### Changelog

**Changed**

- packages/components/src/components/content-switcher/_content-switcher.scss

#### Testing / Reviewing

No testing, just visual testing with storybook knobs

### CODE REVIEW

**Comment by johnbister** [comment]
I'm getting a 'can't handle controls' error in the storybook, but the motion in the recording looks great to me! Do you see this being a one-off animation for  he content switcher or do you see it working for more, similar components (like tab for example)?

@mjabbink Do you have any concerns?

**Comment by AlexanderMelox** [comment]
@johnbister I will get back to you after our micro-interaction meeting on thursdays

**Comment by jnm2377** [comment]
@AlexanderMelox can you sign the DCO? just comment on this PR like it says here: https://github.com/carbon-design-system/carbon/pull/10148#issuecomment-974273998

@johnbister the storybook preview is working for me, here's the link I'm looking at: https://deploy-preview-10148--carbon-components-react.netlify.app/?path=/story/components-contentswitcher--default Maybe you need to clear your cache? 👀

**Comment by AlexanderMelox** [comment]
I have read the DCO document and I hereby sign the DCO.

**Comment by AlexanderMelox** [comment]
Also, I just noticed the preview does not show the animation, but locally (since we are running `start:v11`) works. I think this is because I have to add it to `packages/styles` too ?

**Comment by AlexanderMelox** [comment]
@jnm2377 I saw that you enabled auto-merge. Can we actually disable it until my final review with my motion team? I don't want to redo another PR to fix something.

**Comment by jnm2377** [comment]
@AlexanderMelox I can disable it. FYI It won't merge anything until it's met all the necessary requirements to merge. Which is 2 dev reviews and 1 design review, and all tests passing.

**Comment by jnm2377** [comment]
Yes, the styles need to be added in both v10 and v11 packages. @AlexanderMelox

**Comment by jnm2377** [comment]
@johnbister @AlexanderMelox any updates on this?

**Comment by AlexanderMelox** [comment]
Our motion designer is going to do QA on the animation (any last minute tweaks) and adding the styles to the styles package today

File: packages/styles/scss/components/content-switcher/_content-switcher.scss
Comment (jnm2377) [inline packages/styles/scss/components/content-switcher/_content-switcher.scss]: should be just `motion(standard, productive)` not `carbon--motion(standard, productive)`

Comment (jnm2377) [reply packages/styles/scss/components/content-switcher/_content-switcher.scss]:
```suggestion
      transition: all $duration-moderate-01 motion(standard, productive);
```

File: packages/components/src/components/content-switcher/_content-switcher.scss
Comment (jnm2377) [inline packages/components/src/components/content-switcher/_content-switcher.scss]:
```suggestion
       motion(standard, productive);
```

**Comment by AlexanderMelox** [comment]
I still can't figure out how to make the changes show up on the storybook for v11 and v10. I remember we were running into a bunch of problems when installing packages

**Comment by AlexanderMelox** [comment]
After pulling in some changes from the main branch, I am unable to get my CSS changes to show up in the storybook. I have tried running the storybook both from `packages/carbon-react` and `packages/react` and my changes don't appear.

File: packages/components/src/components/content-switcher/_content-switcher.scss
Comment (janhassel) [inline packages/components/src/components/content-switcher/_content-switcher.scss:64]:
```suggestion
      background-color: $layer-selected-inverse;
```

Comment (janhassel) [inline packages/components/src/components/content-switcher/_content-switcher.scss]:
```suggestion
      background-color: $layer-selected-inverse;
```

File: packages/components/src/components/content-switcher/_content-switcher.scss
Comment (janhassel) [inline packages/components/src/components/content-switcher/_content-switcher.scss]:
```suggestion
    &::after {
```

File: packages/styles/scss/components/content-switcher/_content-switcher.scss
Comment (janhassel) [inline packages/styles/scss/components/content-switcher/_content-switcher.scss]:
```suggestion
      background-color:  $layer-selected-inverse;
```

File: packages/styles/scss/components/content-switcher/_content-switcher.scss
Comment (janhassel) [inline packages/styles/scss/components/content-switcher/_content-switcher.scss]:
```suggestion
    &::after {
```

File: packages/components/src/components/content-switcher/_content-switcher.scss
Comment (janhassel) [reply packages/components/src/components/content-switcher/_content-switcher.scss]: To cover the case of a disabled content switcher, add a new rule like so:

```scss
&:disabled::after {
  display: none;
}
```

File: packages/styles/scss/components/content-switcher/_content-switcher.scss
Comment (janhassel) [reply packages/styles/scss/components/content-switcher/_content-switcher.scss]: To cover the case of a disabled content switcher, add a new rule like so:

```scss
&:disabled::after {
  display: none;
}
```

**Review by janhassel — COMMENTED** [review]
Hey @AlexanderMelox! I think the issue might be due to your selector specificity. Can you try it with these changes?

File: packages/styles/scss/components/content-switcher/_content-switcher.scss
Comment (AlexanderMelox) [reply packages/styles/scss/components/content-switcher/_content-switcher.scss]: I can try these changes, but it was working before this way, nothing in carbon's elements has changed so I don't know why the changes can't be seen in storybook. (I think carbon changed the v10 and v11 styles and which packages pertain to which)

Comment (janhassel) [reply packages/styles/scss/components/content-switcher/_content-switcher.scss]: From what I can see the styles are loaded in correctly, but the selector `&::after:not(&:disabled)` doesn't match. Changing it to `&:not(&:disabled)::after` works but is higher in specificity than `&::after` on line 200 meaning the `scaleY(1)` will not be shown. Hence I suggested to simplify the first selector and create another to address the disabled state.

I'm not 100% sure but the first selector might just be invalid since it would query the `after` pseudo element not being the disabled parent. Did you use this exact selector before?

**Comment by mjabbink** [comment]
So is this the latest preview? If it is I like it! Snappy.
https://deploy-preview-10148--carbon-components-react.netlify.app/?path=/story/components-contentswitcher--default

**Comment by AlexanderMelox** [comment]
@mjabbink Haha, it's not applied on that preview. That's the default

**Comment by mjabbink** [comment]
@AlexanderMelox Where can I see the preview then?

File: packages/components/src/components/content-switcher/_content-switcher.scss
Comment (AlexanderMelox) [reply packages/components/src/components/content-switcher/_content-switcher.scss]: Awesome! For some reason my old changes worked with the not selector, but magically it stopped working. But thank you for this @janhassel !

File: packages/styles/scss/components/content-switcher/_content-switcher.scss
Comment (AlexanderMelox) [reply packages/styles/scss/components/content-switcher/_content-switcher.scss]: I did use that exact same selector and worked fine, that's why i found it weird. But thanks to your suggestions it works now

**Comment by AlexanderMelox** [comment]
@mjabbink I think I just solved the issue! Just pushed so the new preview link should be live soon

**Comment by AlexanderMelox** [comment]
@mjabbink here's the new preview, animations work now 
https://deploy-preview-10148--carbon-components-react.netlify.app/?path=/story/components-contentswitcher--default

**Comment by mjabbink** [comment]
thanks @AlexanderMelox Have you looked at this animation in context of a UI? I like this but I also wonder if it’s too much for what it is? It certainly get’s a visual emphasis and maybe that’s a good thing in context of all the other UI elements/components.

**Comment by jnm2377** [comment]
@johnbister do you have feedback on the animation? on the dev side the code looks good, but this needs design review/approval before it could be merged.

**Comment by johnbister** [comment]
@jnm2377 The motion looks great to me, but I have some concerns about how this fits into the larger context of a UI and our other components like I mentioned on 11/26. It seems @mjabbink has some concerns there as well.  That being said, I am happy to approve it and let's see how it influences motion in the future.

**Review by johnbister — APPROVED** [review]

**Comment by johnbister** [comment]
Thanks for the reminder @jnm2377 :smile:

**Comment by jnm2377** [comment]
Hmm... @johnbister we don't have to approve the PR if you and Mike still have questions around it. Maybe we should close this PR until @AlexanderMelox designers have some more answers/prototypes for our design team to give feedback on. It seems like the 2 big questions are:

1. how and if does this motion work for similar components like Tabs?
2. how does this motion look/work in the context of a full UI? 

I think these are good discussions to be had in the issue, and if/once we reach an agreement then we can re-open the PR?

**Comment by johnbister** [comment]
I stand by my decision to approve it. Great work @AlexanderMelox

**Comment by AlexanderMelox** [comment]
Here's a prototype in the context of a UI
https://codesandbox.io/s/content-switcher-motion-carbon-vanilla-forked-pmicq?file=/src/index.js

I don't think it's too much but can add that breathe of fresh air to some stale pages, especially while switcher contents.

I got the green light from my team to merge this PR ✅

**Comment by johnbister** [comment]
@AlexanderMelox Thanks for sending the example--looks cool! I can definitely see how this might influence other components so we'll be careful to see how it plays out. Just a thought for down the road, but this kind of motion would look great for a dark/light theme switcher, too.

**Review by jnm2377 — APPROVED** [review]

**Comment by AlexanderMelox** [comment]
@abbeyhrt when you have the time, can you review this PR so it can be merged? Thanks!

**Review by abbeyhrt — APPROVED** [review]
Sorry for the delay but it looks good to me! Thank you for the contribution!

**Comment by mjabbink** [comment]
Looks good @AlexanderMelox

**Pull request merged**
